app/bundles/PageBundle/Entity/PageRepository.php

```diff
@@ -14,15 +14,15 @@ class PageRepository extends CommonRepository
      */
     public function getEntities(array $args = [])
     {
-        //use a subquery to get a count of submissions otherwise doctrine will not pull all of the results
         $sq = $this->_em->createQueryBuilder()
             ->select('count(fs.id)')
             ->from('MauticFormBundle:Submission', 'fs')
             ->where('fs.page = p');
 
         $q = $this
             ->createQueryBuilder('p')
-            ->select('p, ('.$sq->getDql().') as submission_count')
+            // submission_count can be used in the order by clause without having it in the result
+            ->select('p, ('.$sq->getDql().') as HIDDEN submission_count')
```

Comment:
Hi @escopecz, i thought about this solution but use this new column is not a retroactive solution, old submissions will not be counted

Reply:
The count won't be visible. It is only necessary to make the sorting work

![image](https://user-images.githubusercontent.com/96085911/190267247-bf33173a-c5ac-4331-afd0-3aa14b6db348.png)

Reply:
@alfredoct96 that's what migrations are for.

@volha-pivavarchyk where is the count in the Results column coming from then?

Reply:
In my view, creating a new column is a good solution in term of performnce. Just one thing. It comes to duplication of data because we already have relation beetwen these tables, and a query gives the same result.
How about adding a new method to PageRepository that will count submissions.

Reply:
> @volha-pivavarchyk where is the count in the Results column coming from then?

https://github.com/mautic/mautic/blob/765f8ccc004408ce671b7e7268dd00e58d8fd59b/app/bundles/PageBundle/Views/Page/list.html.php#L99

Reply:
Yep, it's called [denormalization](https://en.wikipedia.org/wiki/Denormalization). We have the same on list of emails or segments where the table shows different counts of related entities. It's not an easy task to maintain for larger databases and we spent a lot of time to make it fast and accurate on those 2 places. Here are some bugs that we were fixing:

1. The counts cannot be live (meaning making the count query on the relational tables) otherwise the page is slow to load or won't load at all due to HTTP request time limit.
2. We've tried to load them one by one by AJAX but that is a small DDoS to the servers when 100 requests starts a count query at the same time. And users will notice the flashing or changing of the counts after the page loads.
3. We've tried to update the denormalized count on insert and then ensure the inserts are atomic. It slows down the write operations as it must wait for the locks that are needed for atomic writes.
4. We've tried to update the count in a background job at a 10 minute interval but we get support tickets from our customer saying that the count on the list and detail pages is different for a small time period.

So we've done a lot of development work to make such feature work and it still causes frustration for us, our servers and customers.

I hope you understand why I suggest to revert the feature until it is built the way it won't cause frustration. And the feature was built the way it will slow down other parts of the application that do not need form submission counts at all.

Reply:
@volha-pivavarchyk I see now that it's coming from https://github.com/mautic/mautic/pull/11435/files#diff-132641ef0f1cea3ab48d4f2d9ab8f264020e30b052b41536b5615c869b452641R120

That is even less performant though. The count query is now performed in multiple places. Once for sorting and then each row will perform a separate query for each landing page. Do you see how it is worse?

Reply:
@escopecz Of course, I see. Sorry, I was just trying to keep functionality. 

In the beginning, I was just trying to solve the problem with global search. I think we should close this PR. Because this PR will not make sense if PR [#9693](https://github.com/mautic/mautic/pull/9693) is reversed.